src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,526 @@
+ *                          .setProxyPassword(yourProxyPass)
+ *                          .build();
+ * }
+ * </pre>
+ *
+ * <h3>Client with a custom SSL socket factory</h3>
+ * <pre>
+ * {@code
+ * CloudantClient client = ClientBuilder.account("yourCloudantAccount")
+ *                          .setUsername("yourUsername")
+ *                          .setPassword("yourPassword")
+ *                          .setAuthenticatedModeSSLSocketFactory(...)
+ *                          .build();
+ * }
+ * </pre>
+ *
+ * <h3>Client with custom connection and read timeouts</h3>
+ * <pre>
+ * {@code
+ * CloudantClient client = ClientBuilder.account("yourCloudantAccount")
+ *                          .setUsername("yourUsername")
+ *                          .setPassword("yourPassword")
+ *                          .setConnectionTimeout(ClientBuilder.TimeoutOption.minutes(1))
+ *                          .setReadTimeout(ClientBuilder.TimeoutOption.minutes(1))
+ *                          .build();
+ * }
+ * </pre>
+ *
+ * @since 2.0.0
+ */
+public class ClientBuilder {
+
+    /**
+     * Default max of 6 connections
+     **/
+    public static final int DEFAULT_MAX_CONNECTIONS = 6;
+    /**
+     * Connection timeout defaults to 5 minutes
+     **/
+    public static final TimeoutOption DEFAULT_CONNECTION_TIMEOUT
+            = TimeoutOption.minutes(5);
+    /**
+     * Read timeout defaults to 5 minutes
+     **/
+    public static final TimeoutOption DEFAULT_READ_TIMEOUT
+            = TimeoutOption.minutes(5);
+
+    private List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
+            <HttpConnectionRequestInterceptor>();
+    private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
+            <HttpConnectionResponseInterceptor>();
+    private String password;
+    private String username;
+    private URL url;
+    private GsonBuilder gsonBuilder;
+    /**
+     * Defaults to {@link #DEFAULT_MAX_CONNECTIONS}
+     **/
+    private int maxConnections = DEFAULT_MAX_CONNECTIONS;
+    private URL proxyURL;
+    private String proxyUser;
+    private String proxyPassword;
+    private boolean isSSLAuthenticationDisabled;
+    private SSLSocketFactory authenticatedModeSSLSocketFactory;
+    private TimeoutOption connectionTimeout = DEFAULT_CONNECTION_TIMEOUT;
+    private TimeoutOption readTimeout = DEFAULT_READ_TIMEOUT;
+
+    /**
+     * Constructs a new ClientBuilder for building a CloudantClient instance to connect to the
+     * Cloudant server with the specified account.
+     *
+     * @param account the Cloudant account name to connect to e.g. "example" is the account name
+     *                for the "example.cloudant.com" endpoint
+     * @return a new ClientBuilder for the account
+     * @throws MalformedURLException if the specified account name forms an invalid endpoint URL
+     */
+    public static ClientBuilder account(String account) throws MalformedURLException {
```

Comment:
Should this really be a MalformedURLException? Constructing a client via an account name should never cause a MalformedURLException, since we are building the URL from the name provided. An IllegalArugmentException would be more appropriate  in the event the Cloudant Account provided cannot form a valid URL.